plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -169,13 +169,13 @@ private List<String> listRemoteSchemaNames(ConnectorSession session)
     public boolean schemaExists(ConnectorSession session, String schemaName)
     {
         BigQueryClient client = bigQueryClientFactory.create(session);
+        DatasetId localDatasetId = client.toDatasetId(schemaName);
 
         // Overridden to make sure an error message is returned in case of an ambiguous schema name
         log.debug("schemaExists(session=%s)", session);
-        String projectId = client.getProjectId();
-        return client.toRemoteDataset(projectId, schemaName)
+        return client.toRemoteDataset(localDatasetId)
                 .map(RemoteDatabaseObject::getOnlyRemoteName)
-                .filter(remoteSchema -> client.getDataset(DatasetId.of(projectId, remoteSchema)) != null)
+                .filter(remoteSchema -> client.getDataset(DatasetId.of(localDatasetId.getProject(), remoteSchema)) != null)
```

Comment:
The project-id is retrieved from `client` and passed to `getDataset` method. How about changing  `getDataset`'s method argument for taking only `remoteSchema`? Same for `toRemoteDataset` method.

Reply:
The `DatasetId` uniquely identifies a dataset, so if `client.getDataset` receive it and use a different project ID (uses only part of the composite key) it would be an incorrect behavior which only coincidentally works because we always use the same project ID here.